training/src/app/chuck-noris/search/search.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { debounceTime, distinctUntilChanged, Subject, takeUntil, tap } from "rxjs";
+import { FormBuilder, FormGroup, Validators } from "@angular/forms";
+
+@Component({
+	selector: 'app-search',
+	templateUrl: './search.component.html',
+	styleUrls: ['./search.component.scss']
+})
+export class SearchComponent implements OnInit {
+	public searchForm: FormGroup;
+	@Output() public generateJokesFromSearch: EventEmitter<string> = new EventEmitter<string>();
+	@Output() public generateJokesFromClick: EventEmitter<string>  = new EventEmitter<string>();
+	public untilDestroy: Subject<void>                             = new Subject<void>();
+
+	constructor(private fb: FormBuilder) {
+	}
+
+	ngOnInit(): void {
+		this.searchForm = this.fb.group({
```

Comment:
too much logic for ngOnInit.
craete a method "initForm" and move it there, and call it